src/loaders/artifacts.js

```diff
@@ -1,32 +1,54 @@
 import DataLoader from 'dataloader';
 import sift from 'sift';
+import { isNil } from 'ramda';
 import ConnectionLoader from '../ConnectionLoader';
 import Artifact from '../entities/Artifact';
 import Artifacts from '../entities/Artifacts';
 
 export default ({ queue }, isAuthed) => {
-  const withUrl = (taskId, runId, artifact) => {
-    const url = isAuthed
-      ? queue.buildSignedUrl(queue.getArtifact, taskId, runId, artifact.name)
-      : queue.buildUrl(queue.getArtifact, taskId, runId, artifact.name);
+  const withUrl = ({ method, taskId, artifact, runId }) => {
+    const hasRunId = !isNil(runId);
 
-    return { ...artifact, url };
+    if (isAuthed) {
+      return {
+        ...artifact,
+        url: hasRunId
+          ? queue.buildSignedUrl(method, taskId, runId, artifact.name)
+          : queue.buildSignedUrl(method, taskId, artifact.name),
```

Comment:
It looks like `isAuthed` comes from `Boolean(request.credentials)` in `src/createContext.js`, which is to say it's true if the user has provided credentials.  However, I don't think we want to build signed URLs for public artifacts, even when credentials are present -- users often copy/paste artifact URLs, and this would likely lead to people copy/pasting time-limited, signed URLs which would (a) have a long `?bewit=..` argument and (b) probably not work after that bewit expires.

Instead, I think it'd be better to look at `artifact.name`, and sign anything which doesn't begin with `public/`.